### PR TITLE
Feat/send mail

### PR DESCRIPTION
**Pull request by RomainMularczyk**
# Add formatting email functions
* Refactor mail content
* Add mail formatting function unit test

# Add mailjet to dependencies
* Add mailjet v3
* Add mailjet v4

# Add air to git ignored files
* air.toml

# Add discord client
* discord client connection with discord bot token

# Refactor email addresses to variables
* Sender email address
* Receiver email address

# Parallelize main.go
* Run server + discord bot in parallel

# Adding function documentation
* Services
* Repositories

# Add services and repositories
* Add discord_register_id service
* Refactor subscribers-to-sessions repositories to populate intermediate table metadata
* Add repository querying sessions by unique str

# Add discord services
* Discord client connection
* Discord /register command
* Fix /register command error handling (when unique str not found in the intermediate table)

# Routing
* Add groups to restrict /subscribe POST route with middlewares

# Middlewares
* Add middleware checking max number of subscribers per session

# Models
* Modify num_subscribers property to `int`

# Fixes
* Change mailjet sender address environment variable name
* Change logs errors flag to remove date
* Typo in mail content

### CODE REVIEW

File: services/subscribe.go [hunk 1]
```diff
@@ -3,17 +3,31 @@ package services
...
+
+  uniqueStr, err := uuid.NewRandom()
+  if err != nil {
+    logs.Output(
+      logs.ERROR,
+      "Error when generating unique string for a subscriber.",
+    )
+  }
+  mail.SendMail("romain.mularczyk@gmail.com", sess.Name, uniqueStr.String())
 	return nil
```
Comment (myriam-epitech) [inline services/subscribe.go:46]: mail

File: services/mail/send_mail.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	. "github.com/mailjet/mailjet-apiv3-go"
+)
+
+func SendMail(address string, session string, uniqueStr string) {
+	mailjetClient := NewMailjetClient(
+      os.Getenv("MAILJET_API_KEY"),
+      os.Getenv("MAILJET_API_SECRET"),
+    )
+	email := &InfoSendMail {
+      FromEmail: "registration@envnum.fr",
```
Comment (myriam-epitech) [inline services/mail/send_mail.go]: mail

**Review by myriam-tmds — APPROVED** [review]

File: services/discord/discord_command.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            Type: discordgo.InteractionResponseChannelMessageWithSource,
+            Data: &discordgo.InteractionResponseData {
+              Content: fmt.Sprintf(
+                `Vous êtes inscrit à la session : **%s** - *%s*. Elle aura lieu le %s.`,
+                sess.Speaker, sess.Name, sess.Date,
+              ),
+            },
+          },
+        )
+        session.GuildMemberRoleAdd(
```
Comment (myriam-epitech) [inline services/discord/discord_command.go:89]: err pas gérée session.GuildMemberRoleAdd ?

File: services/discord/discord_command.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+// to trigger when a user triggers the `/register` command
+func registerSubscriberDiscordIdCallback(
+  session *discordgo.Session, 
+  interaction *discordgo.InteractionCreate,
+) {
+  if interaction.Type == discordgo.InteractionApplicationCommand {
+    options := interaction.ApplicationCommandData().Options
+    for _, opt := range options {
+      switch opt.Type {
+      case discordgo.ApplicationCommandOptionString:
```
Comment (myriam-epitech) [inline services/discord/discord_command.go:54]: il y a qu'un seul case ?

File: services/discord/discord_command.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+)
+
+func DiscordUserRegistrationCommand(
+  discordClient *discordgo.Session,
+) {
+  appCommand := []*discordgo.ApplicationCommand {
+    {
+      Name: "register",
+      Description: "Registrate a new subscriber.",
+      Options: []*discordgo.ApplicationCommandOption {
```
Comment (myriam-epitech) [inline services/discord/discord_command.go:19]: Register

File: services/mail/mail_content.go [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+                background-color: #4169E1;
+                color: #000000;
+                font-size: 1.3em;
+                padding: 10px;
+                border-radius: 10px;
+                text-decoration: none;
+              '
+            >
+              Rejoindre le server
+            </a>
```
Comment (myriam-epitech) [inline services/mail/mail_content.go:46]: serveur

**Review by myriam-epitech — APPROVED** [review]

File: services/discord/discord_command.go (hunk 4 above)
Comment (RomainMularczyk) [reply services/discord/discord_command.go:54]: Oui

**Pull request merged**